src/Model/mutators.ts

```diff
@@ -10,6 +10,8 @@ var RemoveEvent = mutationEvents.RemoveEvent;
 var MoveEvent = mutationEvents.MoveEvent;
 var promisify = util.promisify;
 
+type ValueCallback<T> = ((error: Error | null | undefined, value?: T) => void);
```

Comment:
With this signature, a consumer won't be able to use traditional Node-style callback result handling in a type-safe way:
```js
model.add(doc, (error, id) => {
  if (error) {
    return handleError(error);
  }
  // TS says `id` could be undefined
  console.log(id.length);
});
```

The types for Node's core libraries use a callback function type like:
```
(error: Error | null | undefined, value: T) => void
```
While that isn't strictly correct either, since it lets you use `value` without checking `error`, it matches better with how Node code has traditionally been written. It also only works because Node core libraries themselves aren't implemented in TS.

Unfortunately, TS doesn't have a great way of handling Node-style callbacks in a way that's type-safe for both the function implementation and consumer.
https://stackoverflow.com/questions/42603810/typescript-error-first-callback-typing-noimplicitany-strictnullchecks

That's all to say:
* The current approach is fine if we don't really care about TS consumers using the callback's `id` in the traditional way with type-safety.
* If we do care about that, then it's better to use the same signature Node does for the public API, and do `as any` shenanigans internally to make that work.

Honestly, it's probably fine to ignore, since consumers using the callback form can always get the id from the synchronous return value. Other thoughts welcome.

Reply:
updated with non-optional `value`